wallet/src/hooks/User/index.jsx

```diff
@@ -25,5 +33,88 @@ export const UserContext = React.createContext({
 export const UserProvider = ({ children }) => {
   // const [state, dispatch] = React.useReducer(reducer, initialState);
   const [state, setState] = React.useState(initialState);
-  return <UserContext.Provider value={[state, setState]}>{children}</UserContext.Provider>;
+  const [isSyncComplete, setIsSyncComplete] = React.useState(false);
+
+  const configureMnemonic = async mnemonic => {
+    const signature = await Web3.signMessage(METAMASK_MESSAGE);
+    const passphrase = jsSha3.keccak256(signature);
+    Storage.mnemonicSet(await Web3.getAccount(), mnemonic, passphrase);
+    setState(previousState => {
+      return {
+        ...previousState,
+        mnemonic,
+      };
+    });
+  };
+
+  const syncState = async () => {
+    const mnemonicExists = Storage.mnemonicGet(await Web3.getAccount());
+    if (mnemonicExists) {
+      const signature = await Web3.signMessage(METAMASK_MESSAGE);
+      const passphrase = jsSha3.keccak256(signature);
+      const mnemonic = Storage.mnemonicGet(await Web3.getAccount(), passphrase);
+      setState(previousState => {
+        return {
+          ...previousState,
+          mnemonic,
+        };
+      });
+    }
+  };
+
+  const setupWebSocket = () => {
+    if (!state.zkpKeys) return;
+    const socket = new WebSocket(optimistWsUrl);
+
+    // Connection opened
+    socket.addEventListener('open', function () {
+      console.log(`Websocket is open`);
+      socket.send('proposedBlock');
+    });
+
+    // Listen for messages
+    socket.addEventListener('message', async function (event) {
+      console.log('Message from server ', event.data);
+      await blockProposedEventHandler(JSON.parse(event.data), state.zkpKeys.ivk, state.zkpKeys.nsk);
+    });
+    setState(previousState => {
+      return {
+        ...previousState,
+        socket,
+      };
+    });
+  };
+
+  const configureZkpKeys = async mnemonic => {
+    if (!mnemonic) return;
+    const zkpKeys = await generateKeys(
+      mnemonic,
+      `m/44'/60'/0'/${DEFAULT_NF_ADDRESS_INDEX.toString()}`,
+    );
+    setState(previousState => {
+      return {
+        ...previousState,
+        zkpKeys,
+      };
+    });
+  };
+
+  React.useEffect(async () => {
+    await syncState();
+    setIsSyncComplete({ isSyncComplete: true });
+  }, []);
+
+  React.useEffect(() => {
+    setupWebSocket();
+  }, [state.zkpKeys]);
+
+  React.useEffect(() => {
+    configureZkpKeys(state.mnemonic);
+  }, [state.mnemonic]);
+
+  return (
+    <UserContext.Provider value={[state, setState, configureMnemonic]}>
+      {isSyncComplete ? children : <div> Signing to MetaMask </div>}
```

Comment:
Replace or comment out this check for now. It's a good idea but we should plan ahead with a loading modal rather than just a white page with text.

Reply:
commented out for now